library/src/androidTest/java/org/xmtp/android/library/GroupTest.kt

```diff
@@ -0,0 +1,204 @@
+        boClient.conversations.newConversation(alix.walletAddress.lowercase())
+        val convos = boClient.conversations.list(includeGroups = true)
+        assertEquals(convos.size, 3)
+    }
+
+    @Test
+    fun testCanSendMessageToGroup() {
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+        group.send("howdy")
+        group.send("gm")
+        assertEquals(group.messages().first().body, "gm")
+        assertEquals(group.messages().size, 3)
+
+        val sameGroup = alixClient.conversations.listGroups().last()
+        assertEquals(sameGroup.messages().size, 2)
+        assertEquals(sameGroup.messages().first().body, "gm")
+    }
+
+    @Test
+    fun testCanSendContentTypesToGroup() {
+        Client.register(codec = ReactionCodec())
+
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+        group.send("gm")
+        val messageToReact = group.messages()[0]
+
+        val reaction = Reaction(
+            reference = messageToReact.id,
+            action = ReactionAction.Added,
+            content = "U+1F603",
+            schema = ReactionSchema.Unicode
+        )
+
+        group.send(content = reaction, options = SendOptions(contentType = ContentTypeReaction))
+
+        val messages = group.messages()
+        assertEquals(messages.size, 3)
+        val content: Reaction? = messages.first().content()
+        assertEquals("U+1F603", content?.content)
+        assertEquals(messageToReact.id, content?.reference)
+        assertEquals(ReactionAction.Added, content?.action)
+        assertEquals(ReactionSchema.Unicode, content?.schema)
+    }
+
+    @Test
+    fun testCanStreamGroupMessages() = kotlinx.coroutines.test.runTest {
+        val group = boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+
+        group.streamMessages().test {
+            group.send("hi")
+            assertEquals("hi", awaitItem().body)
+            awaitComplete()
+        }
+    }
+
+    @Test
+    fun testCanStreamGroups() = kotlinx.coroutines.test.runTest {
+        boClient.conversations.streamGroups().test {
+            val conversation =
+                boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+            conversation.send(content = "hi")
+            assertEquals("hi", awaitItem().messages().first().body)
+            awaitComplete()
+        }
+    }
+
+    @Test
+    fun testCanStreamGroupsAndConversations() = kotlinx.coroutines.test.runTest {
+        boClient.conversations.stream(includeGroups = true).test {
+            val group =
+                boClient.conversations.newGroup(listOf(alix.walletAddress.lowercase()))
+            val conversation =
+                boClient.conversations.newConversation(alix.walletAddress.lowercase())
+            assertEquals("hi", awaitItem().messages().first().body)
+            awaitComplete()
+        }
+    }
```

Comment:
Non of my streaming tests pass... so I'm either doing the tests wrong or I'm setting up streaming wrong.